virtcontainers/qemu.go

```diff
@@ -1074,33 +1075,61 @@ func (q *qemu) hotplugMemory(memDev *memoryDevice, op operation) (int, error) {
 		return 0, fmt.Errorf("cannot hotplug negative size (%d) memory", memDev.sizeMB)
 	}
 
-	// We do not support memory hot unplug.
-	if op == removeDevice {
-		// Dont fail for now, until we fix it.
-		// We return that we only unplugged 0
-		q.Logger().Warn("hot-remove VM memory not supported")
-		return 0, nil
-	}
-
+	q.Logger().WithField("hotplug-memory-mb", memDev.sizeMB).Debug("hotplugging memory")
 	err := q.qmpSetup()
 	if err != nil {
 		return 0, err
 	}
 
-	maxMem, err := q.hostMemMB()
-	if err != nil {
-		return 0, err
+	currentMemory := int(q.config.MemorySize) + q.state.HotpluggedMemory
+
+	if memDev.sizeMB == 0 {
+		q.Logger().WithField("hotplug", "memory").Debug("hotplug is not required")
+		return 0, nil
 	}
 
-	// calculate current memory
-	currentMemory := int(q.config.MemorySize) + q.state.HotpluggedMemory
+	switch op {
+	case removeDevice:
+		q.Logger().WithFields(
+			logrus.Fields{
+				"hotplug":   "memory",
+				"operation": "remove",
+			},
+		).Debugf("Requested to remove memory: %d MB", memDev.sizeMB)
+		// Dont fail for now, until we fix it.
+		// We return that we only unplugged 0
+		q.Logger().Warn("hot-remove VM memory not supported")
+		return 0, nil
+	case addDevice:
+		q.Logger().WithFields(
+			logrus.Fields{
+				"hotplug":   "memory",
+				"operation": "add",
+			},
+		).Debugf("Requested to add memory: %d MB", memDev.sizeMB)
+		maxMem, err := q.hostMemMB()
+		if err != nil {
+			return 0, err
+		}
 
-	// Don't exceed the maximum amount of memory
-	if currentMemory+memDev.sizeMB > int(maxMem) {
-		return 0, fmt.Errorf("Unable to hotplug %d MiB memory, the SB has %d MiB and the maximum amount is %d MiB",
-			memDev.sizeMB, currentMemory, q.config.MemorySize)
+		// Don't exceed the maximum amount of memory
+		if currentMemory+memDev.sizeMB > int(maxMem) {
+			// Fixme: return a typed error
```

Comment:
I guess you'll fix it in this pr

Reply:
I am not fixing in this PR most of our errors are not typed today, would be good to have a work for that.